python: Add warning message for set_workload being called twice

erin-le opened the pull request:
This commit adds a warning message for when set_workload is called twice, as users typically do not mean to do this.

erin-le committed:
python: Add warning message for set_workload being called twice

This commit adds a warning message for when set_workload is called
twice.

Change-Id: Ifaa0aa37dce14a1a0465c801e788f5ebbcdd8fa2

erin-le committed:
python: fix import in kernel_disk_workload.py

This commit restores an import that was accidentally removed
when adding the error message.

Change-Id: I22fab8272ed7bc92d5bfcd606f899b72568ac164

Harshil2107 commented on src/python/gem5/components/processors/base_cpu_core.py:
```diff
@@ -135,6 +135,11 @@ def connect_walker_ports(self, port1: Port, port2: Port) -> None:
     def set_workload(self, process: Process) -> None:
         self.core.workload = process
 
+    def is_workload_set(self) -> bool:
```
It might be useful to move this function to `AbstractCore` so that anyone who makes their own cores would also have access to this function.

Harshil2107 commented on src/python/gem5/components/boards/kernel_disk_workload.py:
```diff
@@ -187,6 +188,9 @@ def set_kernel_disk_workload(
         # full-system simulation.
         self._set_fullsystem(True)
 
+        if self.workload.object_file:
```
You might be able to get access to `is_workload_set()` function if it is moved to `AbstractCore` as the `AbstractBoard` has an `AbstractProcessor` which has the `AbstractCore`, so it might be possible to use the same method to do this check as done in the `set_se_binary_workload` function.

erin-le committed:
python: change set workload twice warning to use flag

Harshil2107 commented on src/python/gem5/components/boards/kernel_disk_workload.py:
```diff
@@ -74,6 +75,8 @@ def get_default_kernel_args(self) -> List[str]:
         * This assumes the Linux kernel is used.
     """
 
+    _is_workload_set = False
```
I think it would be better to move this to `abstract_board` with functions to get the value and set it like we do with the `_is_fs` function. This will move the logic to one single place instead of having to write the logic in every set function that sets workloads.

erin-le committed:
python: move _is_workload_set into AbstractBoard

This commit moves _is_workload_set, the flag for determining if a
workload had been set for a second time, into AbstractBoard
instead of having it be in kernel_disk_workload and
se_binary_workload

Harshil2107 reviewed (dismissed):
lgtm

powerjg commented on src/python/gem5/components/boards/se_binary_workload.py:
```diff
@@ -141,6 +145,7 @@ def set_se_binary_workload(
             #
             # A better API for this which avoids `isinstance` checks would be
             # welcome.
+
```
Spurious whitespace change

erin-le committed:
python: remove whitespace change
